test/async-hooks/test-promise.chain-promise-before-init-hooks.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const initHooks = require('./init-hooks');
+const { checkInvocations } = require('./hook-checks');
+
+const p = (new Promise(common.mustCall(executor)));
+p.then(afterresolution);
+
+// init hooks after chained promise is created
+const hooks = initHooks();
+hooks._allowNoInit = true;
+hooks.enable();
+
+function executor(resolve, reject) {
+  resolve(5);
+}
+
+function afterresolution(val) {
+  assert.strictEqual(val, 5);
+  return val;
+}
```

Comment:
I think this would be a bit more readable if you just inlined `executor` and `afterresolution`